exporter/awsxrayexporter/translator/segment.go

```diff
@@ -326,30 +326,49 @@ func timestampToFloatSeconds(ts pdata.TimestampUnixNano) float64 {
 	return float64(ts) / float64(time.Second)
 }
 
-func sanitizeAndTransferAnnotations(dest map[string]interface{}, src map[string]string) {
-	for key, value := range src {
-		key = fixAnnotationKey(key)
-		dest[key] = value
-	}
-}
-
-func makeAnnotations(attributes map[string]string) (string, map[string]interface{}) {
+func makeXRayAttributes(attributes map[string]string, indexedAttrs []string, indexAllAttrs bool) (
+	string, map[string]interface{}, map[string]map[string]interface{}) {
 	var (
-		result = map[string]interface{}{}
-		user   string
+		annotations     = map[string]interface{}{}
+		metadata        = map[string]map[string]interface{}{}
+		user            string
 	)
 	delete(attributes, semconventions.AttributeComponent)
 	userid, ok := attributes[semconventions.AttributeEnduserID]
 	if ok {
 		user = userid
 		delete(attributes, semconventions.AttributeEnduserID)
 	}
-	sanitizeAndTransferAnnotations(result, attributes)
 
-	if len(result) == 0 {
-		return user, nil
+	if len(attributes) == 0 {
+		return user, nil, nil
 	}
-	return user, result
+
+	if indexAllAttrs {
+		for key, value := range attributes {
+			key = fixAnnotationKey(key)
+			annotations[key] = value
+		}
+	} else {
+		defaultMetadata := map[string]interface{}{}
+		for key, value := range attributes {
+			found := false
+			for _, indexedKey := range indexedAttrs {
```

Comment:
Instead of n*m, let's put indexedAttrs into a map (no set in Go it seems) and lookup from it to check

Reply:
Good idea